packages/core/client.js

```diff
@@ -14,28 +14,29 @@ const REPORT_USAGE_ERR_PREFIX = `Bugsnag usage error.`
 const noop = () => {}
 
 class BugsnagClient {
-  constructor (notifier, configSchema = config.schema, session = null) {
+  constructor (notifier) {
     if (!notifier || !notifier.name || !notifier.version || !notifier.url) {
       throw new Error('`notifier` argument is required')
     }
 
     // notifier id
     this.notifier = notifier
 
-    // config
-    this.configSchema = configSchema
-
     // configure() should be called before notify()
     this._configured = false
 
-    // i/o
+    // intialise opts and config
+    this._opts = {}
+    this.config = {}
+
+    // // i/o
     this._delivery = { sendSession: noop, sendReport: noop }
     this._logger = { debug: noop, info: noop, warn: noop, error: noop }
 
     // plugins
-    this.plugins = []
+    this._plugins = {}
 
-    this.session = session
+    this._session = null
```

Comment:
do we need to worry about backward compatibility for this one at all?

Reply:
Should be fine. It's an internal API only used by us, which is why I decided to prefix with the pseudo-private (by convention) underscore to denote as such.